core/utils.py

```diff
@@ -84,6 +84,47 @@ def parse_bool(string: str) -> Optional[bool]:
             return False
         return None
 
+    @staticmethod
+    def create_table(iterable: Iterable, attributes: Dict[str, str]) -> str:
+        """Create table from any iterable.
+
+        This is useful mainly for '<command> list' situations.
+
+        Args:
+            iterable: Any iterable of objects to create the table from.
+            attributes: Dictionary of attributes and their translations.
+        """
+        matrix: List[List[str]] = []
+
+        # header
+        matrix.append(list(attributes.values()))
+        # body
+        for item in iterable:
+            line: List[str] = []
+            for attr in attributes.keys():
+                line.append(str(getattr(item, attr, "")))
+            matrix.append(line)
+
+        # compute column widths
+        widths: List[int] = [0 for i in range(len(attributes))]
+        for line in matrix:
+            for i in range(len(attributes)):
+                width = len(line[i])
+                if widths[i] < width:
+                    widths[i] = width
```

Comment:
You could squash this into the first for loop right? Saving those precious miliseconds.

Reply:
I was worried it would get harder to read. Will try though.